Add page to upload dd214 or other supporting docs

erikphansen opened the pull request:
Hide the page if the user is in MVI:
- if they are logged in, we can check this via their profile in the
redux store
- if they are not logged in, we can determine this based on the response
from the ID Form they had to fill out

## Description


## Testing done


## Screenshots
Before uploading anything:
![image](https://user-images.githubusercontent.com/20728956/54149921-8db45e00-43f4-11e9-9e04-1a93ab7b7592.png)

Uploaded file details:
![image](https://user-images.githubusercontent.com/20728956/54149931-90af4e80-43f4-11e9-99b6-9cd7b814b347.png)

Document type options:
![image](https://user-images.githubusercontent.com/20728956/54149945-95740280-43f4-11e9-984c-cd64be27889c.png)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

erikphansen committed:
Add page to upload dd214 or other supporting docs

Hide the page if the user is in MVI:
- if they are logged in, we can check this via their profile in the
redux store
- if they are not logged in, we can determine this based on the response
from the ID Form they had to fill out

erikphansen committed:
hide dd214 upload page on production

erikphansen commented on src/applications/hca/config/form.js:
```diff
@@ -33,23 +35,24 @@ import GetFormHelp from '../components/GetFormHelp';
+  medicalCentersByState,
+  medicarePartADescription,
+  prefillTransformer,
+  transform,
```
sorry for the noise, just putting these in alpha order. the only real change was adding the `prefillTransformer`

erikphansen committed:
cleanup

erikphansen commented on src/applications/hca/config/form.js:
```diff
@@ -84,70 +88,144 @@ const emptyObjectSchema = {
 const {
-  gender,
-  mothersMaidenName,
+  campLejeune,
```
more reorganizing in alpha order

erikphansen committed:
move fileHelp from form config to form helpers

erikphansen committed:
update path

erikphansen committed:
Merge branch 'master' into dd214-upload

U-DON commented on src/applications/hca/helpers.jsx:
```diff
@@ -98,6 +134,47 @@ export const facilityHelp = (
+  }
+
+  return (
+    <React.Fragment>
```
Could use the shorthand here.

U-DON commented on src/applications/hca/helpers.jsx:
```diff
@@ -98,6 +134,47 @@ export const facilityHelp = (
 
+export function fileHelp({ formContext }) {
+  if (formContext.reviewMode) {
+    return <p />;
```
Is it possible to just return `null`, or does that do something weird?

erikphansen replied:
Cool, `null` seems to work. I had just copied this from another implementation of the file uploader.

U-DON commented on src/applications/hca/helpers.jsx:
```diff
@@ -14,9 +14,40 @@ import {
 
 import facilityLocator from '../facility-locator/manifest';
 
+export function prefillTransformer(pages, formData, metadata, state) {
```
Definitely seems like a better way to set the view field. With this, I think you can remove the obsolete code from `IDPage`.

erikphansen replied:
thanks for catching that. I might need to rework the redirection logic on the `IDPage` now...

U-DON reviewed (approved):
LGTM.

erikphansen committed:
Work in PR feedback.

Also rework handling of enrollement_status error that indicates the user
was not in MVI.

erikphansen commented on src/applications/hca/actions.js:
```diff
@@ -9,7 +10,13 @@ export function submitIDForm(formData) {
     dispatch({ type: SUBMIT_ID_FORM_STARTED });
 
-    const baseUrl = '/health_care_applications/enrollment_status';
+    // We need bypass the local API when running locally, so hit dev-api
```
I'm not sure if this is something we have done in the past, but this seems to be better than manually changing the value of `baseUrl` when running locally and wanting to get a non-500 response from the `enrollement_status` endpoint.

rianfowler commented on src/applications/hca/config/form.js:
```diff
@@ -520,6 +557,47 @@ const formConfig = {
+              addAnotherLabel: 'Upload another document',
+              fileUploadUrl: `${environment.API_URL}/v0/hca_attachments`,
+              fileTypes: ['pdf', 'doc', 'docx', 'jpg', 'jpeg', 'rtf', 'png'],
+              maxSize: 1024 * 1024 * 10,
```
The description suggests this is a cumulative limit. I think maxSize applies to the single file. Do we know what happens when they upload more than 10MB of files? 

See: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14122

erikphansen replied:
good catch. yeah this is a known issue that we aren't worried about at this point since it's such an edge case. but sadly at this point there is no way to show errors when the _total_ upload size is too large. and if someone does upload multiple files that exceed the backend's limit (which is actually closer to 15MB) then the submission will fail and it will be a horrible UX :(

If I made a ticket to enhance the file upload ui widget to add this support should I tag your team on that? I'm not sure who would be responsible for an enhancement like that.

rianfowler replied:
yeah you can tag our team since it's a `forms-system` enhancement. this will be the second form I'm aware of that needs a cumulative size limit enforced.

erikphansen committed:
resolve merge conflicts
